### DISCUSSION TITLE
[Suggestions] Creature separation window (and others windows), allow the use of the mouse wheel anywhere on the screen

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#2684] [Ideas]
In fheroes2, when this window is open, it would be nice to allow the use of the mouse wheel when the mouse cursor is located anywhere on the screen and not only when the mouse cursor is positioned just over the number of creatures indicator.

Same for the number of creatures to recruit in the Creature recruit window.
Same for the number of resources to trade in the Marketplace window.

In fheroes2:

![Separation slots fh2 avec boutons 3 4 5 01](https://user-images.githubusercontent.com/43583994/106055462-b24f7700-60ed-11eb-85b7-9d6b6ae775f7.png)

![Recruit window mouse wheel fh2 01](https://user-images.githubusercontent.com/43583994/106059263-c0ec5d00-60f2-11eb-9e34-053de557108b.png)

![Marketplace window mouse wheel fh2 01](https://user-images.githubusercontent.com/43583994/106059284-c9dd2e80-60f2-11eb-8a43-7ea8f6599f64.png)

**Edit 2021/02/23.**

It should also be possible to use the mouse wheel to change pages in the Magic Book (adventure and combat spells, on the adventure map and during a combat).

![Mouse wheel magic book 01](https://user-images.githubusercontent.com/43583994/108828135-c2605680-75c6-11eb-9e02-491f86ddf6aa.png)

**Edit 2021/09/21.**

In the Kingdom Overview screens, allow the use of the mouse wheel for lists on the whole screen (not only on the upper part of the screen corresponding to the list area).

In fheroes2:

![Allow mouse wheel whole screen 01](https://user-images.githubusercontent.com/43583994/134159548-efeeb9fb-024d-442f-89c1-4176c739e916.png)

And other screens when possible.

In fheroes2:

Experimental Game Settings screen:
![Experimental wheel whole screen 01](https://user-images.githubusercontent.com/43583994/134159678-4d475336-725f-430c-97c4-25be764f628a.png)

Town Portal screen:
![Town Portal wheel whole screen 01](https://user-images.githubusercontent.com/43583994/134159704-6434c3c0-3158-4f29-965c-e2e9bdd1140f.png)

Hero screen:
![Hero wheel whole screen 01](https://user-images.githubusercontent.com/43583994/134161137-e19a1a52-3bd2-425f-9bc9-8b6b78efaa1e.png)

Castle screen:
![Castle wheel whole screen 01](https://user-images.githubusercontent.com/43583994/134161161-04382855-ae94-4214-b08c-6fc1c9955c65.png)

**Comment by Branikolog** [comment]
Hi, @LeHerosInconnu.
I agree, that we should expand the scrolling area on the whole screen for each case you performed here.
Also, I think for the issue, we've spammed before too. :D (#2420)
Well, actually, I think scrolling should work everywhere in the game the same, as you proposed here. I believe, every time we watch quantity box or scroll bar on a screen should be the case. except world map window, as there we have two scroll bars for heroes and towns.
